### PR TITLE
LastUpdate Prefix for News: Change from "Updated on" to "Published on"

### LINKED ISSUE #283: LastUpdate Prefix for News
On news pages only, the hero "Updated on xyz" text prefix shall be changed to "Published on xyz" 

Ref: https://adobe-dx-support.slack.com/archives/C06H8UNQ3RS/p1709913818323329?thread_ts=1708639499.998829&cid=C06H8UNQ3RS

### PR DESCRIPTION
**Pull request by pahupe**
On news pages only, the hero "Updated on xyz" text prefix shall be changed to "Published on xyz"
Ref: https://adobe-dx-support.slack.com/archives/C06H8UNQ3RS/p1709913818323329?thread_ts=1708639499.998829&cid=C06H8UNQ3RS

Fix #283 

Test URLs (News, change expected: Updated on changed to Published on):
- Before: https://main--hlx-test--urfuwo.hlx.page/news/0000/the-article
- After: https://283-lastupdated-prefix--hlx-test--urfuwo.hlx.live/news/0000/the-article

Test URLs (Blog, no change expected: Updated on remains as is):
- Before: https://main--hlx-test--urfuwo.hlx.page/blog/2023/12/being-human-in-the-age-of-ai
- After: https://283-lastupdated-prefix--hlx-test--urfuwo.hlx.live/blog/2023/12/being-human-in-the-age-of-ai

### CODE REVIEW

**Review by mhaack — CHANGES_REQUESTED** [review]
This is the wrong approach, this should not be loaded from metadata, this should be loaded from placeholders. Author should not care about detail.

**Comment by saurabh-khare** [comment]
> This is the wrong approach, this should not be loaded from metadata, this should be loaded from placeholders. Author should not care about detail.

I had the same question but what if editors want to change it only for one page? (That may be going too far anyways)

**Comment by pahupe** [comment]
> This is the wrong approach, this should not be loaded from metadata, this should be loaded from placeholders. Author should not care about detail.

@mhaack , @saurabh-khare: Thanks for the feedback. 

Note: the metadata is loaded from /metadata.xlsx (https://sap.sharepoint.com/:x:/r/sites/207899/Shared%20Documents/website/metadata.xlsx?d=w15bcd449c252491b88f9fc8dae437be3&csf=1&web=1&e=dDjADM), as different values apply to different paths (/news/** uses a different prefix than e.g., /blog/**). Editors don't add this metadata entry to each and every article (but they could). In general, editors don't have to change anything in their existing documents; the path (/news/** or not /news/**) tells what last update prefix should be used (this is how I read Elaine's comment) with "Updated on" being the fallback.

Still, I like the placeholder idea. Should I replace the verbatim string in /metadata.xlsx with a placeholder and then run it through placeholder replacement logic, or is this a level of indirections too many?

**Comment by saurabh-khare** [comment]
> > This is the wrong approach, this should not be loaded from metadata, this should be loaded from placeholders. Author should not care about detail.
> 
> @mhaack , @saurabh-khare: Thanks for the feedback.
> 
> Note: the metadata is loaded from /metadata.xlsx (https://sap.sharepoint.com/:x:/r/sites/207899/Shared%20Documents/website/metadata.xlsx?d=w15bcd449c252491b88f9fc8dae437be3&csf=1&web=1&e=dDjADM), as different values apply to different paths (/news/** uses a different prefix than e.g., /blog/**). Editors don't add this metadata entry to each and every article (but they could). In general, editors don't have to change anything in their existing documents; the path (/news/** or not /news/**) tells what last update prefix should be used (this is how I read Elaine's comment) with "Updated on" being the fallback.
> 
> Still, I like the placeholder idea. Should I replace the verbatim string in /metadata.xlsx with a placeholder and then run it through placeholder replacement logic, or is this a level of indirections too many?

You can add the placeholder in placeholders.xlsx here: https://sap.sharepoint.com/:x:/r/sites/207899/_layouts/15/Doc.aspx?sourcedoc=%7B96A19BCF-C9F4-4928-B566-0BE917C92A98%7D&file=placeholders.xlsx&action=default&mobileredirect=true
Then follow this logic to get the placeholder
https://github.com/urfuwo/hlx-test/blob/d9ced470b9ca051d543ce879d6e64fe4307a0434/aemedge/blocks/related-articles/related-articles.js#L36

**Comment by pahupe** [comment]
Discussion with @mhaack results:
- Remove value from /metadata.xlsx, do distinction by path in hero
- Add a Null Check for lastUpdated

File: aemedge/blocks/hero/hero.js [hunk 1]
```diff
@@ -28,9 +28,13 @@ function decorateMetaInfo() {
   const lastUpdate = getMetadata('modified-time')
     ? getMetadata('modified-time')
     : getMetadata('published-time');
-  infoBlockWrapper.append(
-    span({ class: 'media-blend__date' }, `Updated on ${formatDate(lastUpdate)}`),
-  );
+  if (lastUpdate) {
+    const path = window.location.pathname.split('/');
```
Comment (mhaack) [inline aemedge/blocks/hero/hero.js:32]: `window.location.pathname.indexOf('/news/') > -1` is maybe simpler and can be done in one line

Comment (pahupe) [reply aemedge/blocks/hero/hero.js:32]: I don't want it applied to /some-other-area/news/03/..., just to /news/*

Comment (mhaack) [reply aemedge/blocks/hero/hero.js:32]: Ok than use `window.location.pathname.startsWith('/news/')` still way simpler the splitting and checking the array.

Comment (pahupe) [reply aemedge/blocks/hero/hero.js]: Done.

**Review by mhaack — APPROVED** [review]

**Pull request merged**